### DISCUSSION TITLE
Feedback

### DISCUSSION THREAD

**Discussion by rgerganov** [rgerganov/gen5fw#1] [Announcements]
Use this space for comments, feedback and questions about creating custom firmware for Gen5 cars.

**Comment by quanmltya** [comment]
Dear @rgerganov ,
I have lk.rom of the head unit (Android 4.4) from KIA (Vietnam), could you please investigate if there is any secret code/button/usb file to enter the fastboot?
It is because I did updating the incorrect "update.zip" file and now the unit is in the recovery boot-loop.

Add more information: It is the telechip TCC897x soc
Thanks in advanced!
[lk.rom.zip](https://github.com/rgerganov/gen5fw/files/10799884/lk.rom.zip)

**Comment by DobbyNator94** [comment]
Hey, I have some ideas:

- Would be really nice to have the new UI theme in my Hyundai Ioniq 2019, like here:

But I think the Hyundai Ioniq 2019 electric has the Compact-Class Gen5? Or Standard-Class Gen5? I'm not really 100% sure. Is it possible too? I've looked in my settings and I have the Compact-Claas Gen5.

https://update.hyundai.com/EU/DE/updateNoticeView/1001

![image](https://user-images.githubusercontent.com/40775494/228888855-eb63a33d-937e-49ae-9a19-e76e6167914f.png)
or the new Hyundai theme:
![image](https://user-images.githubusercontent.com/40775494/228888967-73af2582-3360-46f3-b8d5-2ebb3a6031b2.png)

Maybe a rebuild or patch mod for the fw-version with different UI styles would be great?

Or the new Energy Overview from Ioniq 5 etc:

![image](https://user-images.githubusercontent.com/40775494/228889415-ea84c413-85b6-4686-b2cb-a90d5eb49fd2.png)

And here in Europe the Ioniq electric 2016-2019 has no BlueLink, so no possibility to remote start climation. Maybe something is possible, when someone analyze the .APK for scheduled climation. A wifi hotspot is not the biggest problem for the connectivity.

Thank you for work!

**Comment by 395v** [comment]
Is it possible to hack non-navi version? System does not see latest update in the usb btw. Also bootloader can not find required files to update even if I prepared and copied them in a usb flash disk. It is Tucson 2020 tlfl.eur.sop.v095.181008.dau_a

![IMG_6932](https://user-images.githubusercontent.com/65404527/233857071-f76a7021-45a6-4c71-8349-26b1c8c61095.jpeg)

**Reply by Szutek90** [reply]
You are doing something wrong because because i did security_force update on NoNav Device. There are few problems; 1. Software is different for NoNav and Nav, 2 . I cannot find any other software than for USA market.

**Reply by 395v** [reply]
All encrypted files are ready in usb flash. Nothing happens as you can see the screenshot. I prepared v134. Units version is 95. Does nav firmware work on no-nav unit?

**Reply by Szutek90** [reply]
Nav firmware is not working on no-nav devices. For example binding of buttons is differtent.

**Reply by piotr-cz** [reply]
@Szutek90

Did you succeed to hack non-navi version?
There is no firmware available for it in Navigator Updater, at least for EU region

**Comment by patr1x007** [comment]
Hi @rgerganov how are you?
I have a question, do you know how to change option in Engineering mode Variant?
I mean AMP section where is options Internal Amp, External Amp (Mobis), External Amp (Harman).
I try to change options but radio button not change!
Thanks for help

**Reply by patr1x007** [reply]
After some investigation and decompiling, I have some puzzles. But I still don't know what is under "getVariantCode(0x4);"
It looks that I am still missing some references for example device namespace... "com.hkmc.device.VariantCoding;"

EngineerModeConfig.smali:
`    public static int getAmpType() {
        AMP_TYPE = VariantCoding.getVariantCode(0x4);
        Log.e("EngineerModeConfig", "AMP_TYPE : " + AMP_TYPE);
        return AMP_TYPE;
    }`

My goal is working menu in AMP section. I want to run original external krell amplifier, so I need "External Amp (Mobis)" as working option

**Comment by stavultras** [comment]
Hello, how did you obtain partition.dat file? Seems like it doens't work for Tucson TLFL 2018 Korean. I broke ZIP password, did img2simg and gen5.py encrypt fw security_force. Upgraded from USB and now I see only boot logo (Hyundai) ))). I think it's a problem with partition.dat.

Thanks.

**Reply by rgerganov** [reply]
I got partition.dat from an internet forum. I don't know its structure but it worked for Kia Stringer and some Hyundai models. If anyone has some information or can provide another partition file, that would be nice.

**Reply by rgerganov** [reply]
There is some analysis by @adelie93 in https://github.com/rgerganov/gen5fw/discussions/15

**Comment by upintheairsheep** [comment]
Hello, may you please dump the firmware  and upload to the internet archive???

**Reply by upintheairsheep** [reply]
I do not have access to the firmware because you need to own the car.

**Comment by dbolger** [comment]
Hi, @rgerganov

I'm having trouble getting the contents of update_package.zip. I tried using the methods in the blog post but they don't seem to be working for me with the last modification date on the newest firmware update for a 2020 Hyundai Santa Fe (USA/Navigation).

Has anyone lately had any success?  I see the ZIP is still using ZipCrypto but my efforts haven't been fruitful. I may be using the tool wrong, or have the setup wrong.

**Reply by dbolger** [reply]
++ @dracode

**Reply by stavultras** [reply]
I can check your ZIP file, but after Sept 18th. But prepare to have a brick. I already have one :)

**Reply by dbolger** [reply]
@stavultras - If you have Discord, you can add me there - My username is the same as here. If that doesn't work for you, let me know.

**Reply by stavultras** [reply]
@dbolger I have only telegram. The same username.

**Reply by upintheairsheep** [reply]
Can you upload the zip to the Internet Archive please

**Comment by graftonskaggs** [comment]
You know I have a 2021 Hyundai palisade and there seems to be some differences on mine I'm trying to work out. Like the updates aren't exactly the way they used to be, like these ones are .tar files with no passwords but I did find those images in there. But I guess they're supposed to be a header with the file size that isn't in there. Mine are called enc_ instead of encrypt. I can see though from engineering you how to kick it into that mode where it looks for those files. Seems like it's saying cannot find update.fai. I was wondering about if you have any idea how to decrypt or do you have the zip file you used to update because I'll  wanted to see how the images should look.

**Reply by dbolger** [reply]
> You know I have a 2021 Hyundai palisade and there seems to be some differences on mine I'm trying to work out. Like the updates aren't exactly the way they used to be, like these ones are .tar files with no passwords but I did find those images in there. But I guess they're supposed to be a header with the file size that isn't in there. Mine are called enc_ instead of encrypt. I can see though from engineering you how to kick it into that mode where it looks for those files. Seems like it's saying cannot find update.fai. I was wondering about if you have any idea how to decrypt or do you have the zip file you used to update because I'll  wanted to see how the images should look. 

Have you seen this blog?

https://programmingwithstyle.com/

**Reply by graftonskaggs** [reply]
Yeah, its not the same system. I can put a file named update.fai on the usb drive and then it says tag read error so it must be looking for that file. I tried changing the FileInfo file to update.fai but it still says tag read error. That boot loader update process seems promising but its not exactly the same.

**Reply by graftonskaggs** [reply]
This is the one I have https://www.ebay.com/itm/155474068286

**Comment by graftonskaggs** [comment]
Well, just to test this out I managed to put the software from the Tuscan onto a non nav Tuscan which isn't a good idea because the update mechanism stops working. Starts saying rsa key failed. But anyway where did that partition.dat file come from because I'm wondering if it's safe to use that same file on all the updates. Because it might have the data for how the emmc should be partitioned and that might not be the same for every head unit. Pretty cool though, I appreciate the work.

**Comment by AltSyndrome** [comment]
Hi, thanks for all your work, first of all. I have been breaking my head trying to get into engineering mode in the first place after the Nov 2023 update for Kia (see head unit info below). None of the known methods seem to work. 

1. 7-3-1 only gives access to dealer mode with code 2400, no other codes are accepted.
2. Tapping 5/10 times left of the update button, followed by 1/2 times right of the update button doesn't ever lead to a screen to enter a code at all.
3. Tapping 7 times from left to right and back in the screensaver menu is not an option, as the screensaver menu is a 3-item vertical textlist rather than the 3 horizontal images of clock types, so there's no space beneath it to enter the tapping sequence.

Any alternative ideas? Any help is much appreciated.

![IMG_7933 2](https://github.com/rgerganov/gen5fw/assets/155300926/0999eafa-bee7-4a17-b2b3-e263334f135c)
![IMG_7934](https://github.com/rgerganov/gen5fw/assets/155300926/e17bce8f-753f-4e68-a668-c65d3cfb1a1e)

**Comment by fadyv88** [comment]
Hello, I would really appreciate if someone could help
I have a KOR variant on my kia forte, it is a compact GEN5 navigation unit
However, I need to install the update and the maps for the middle east version.
![20240128_104935](https://github.com/rgerganov/gen5fw/assets/157991131/6132c51b-1037-4d69-b2da-e34afb04efe9)

How do I compile it to update or flash the model on the headunit
Your help is so much appreciated

**Reply by Szutek90** [reply]
Its gen 4 not Gen5

**Reply by fadyv88** [reply]
Why are you saying that? In the info under settings it is compact gen5 navigation, the picture is attached.

**Reply by Szutek90** [reply]
Ohh...sorrry, my mistake.

**Reply by fadyv88** [reply]
No worries, do you know how can I get to update from another region? I can't even get the update for korea from where I live (Egypt) and kia support are useless

**Reply by Szutek90** [reply]
Type "Navigation Updater hyundai" in google and you will find. From official Hyundai update system. It's free.

**Reply by fadyv88** [reply]
I  can't download the korean version from outside of Korea. Also, the update is not recognised when trying to update using the mea files on the korean headunit

**Reply by imtmiissa** [reply]
> on the korean headunit

you can use VPN to install it

**Reply by mahmodHammad** [reply]
Hi @fadyv88 
Have out found a solution for this ?

**Reply by hamoodfruit** [reply]
Hi, any update on this? I have a 2020 hyundai sonata with standard class Gen5W navigation and i want to change the USA software to Middle east software to get the middle east maps, navigation, and road speed limits

**Comment by PatrikTrestik** [comment]
@rgerganov Do you know method to get ADB on version 221117 or newest 230911?
They changed structure and there is no update.zip anymore.
SD contains set of files. Some are probably archives but I could not unpack it "npkg".

Any idea?

**Reply by mahmodHammad** [reply]
Same with me
@PatrikTrestik  have you found a solution?
thanks

**Comment by imtmiissa** [comment]
can you shear with me a video for the way

**Comment by Durioner** [comment]
Hey,
I have problem with finding password for Hyundai Ioniq update_package.ziop. I tried with bkcrack and nothing was found. Maybe anyone have ready file for 2021 version of EU software? I will be much grateful

**Comment by claudioaferreira** [comment]
HI, there's something for hyundai tucson 2020 TLFL.USA.SOP.V115.221011 ?
Also in my tucson 2020 i can't change MIles to Killometer...

**Comment by IvanVojtko** [comment]
Hey @rgerganov . How did you manage to connect using ADB? I've enabled ADB but I can't connect over network. I tried USB cable as well, but no device is recognized. Could you help me with this?

**Comment by makarich** [comment]
Hi, there's something for kia sportage V from china? Or how can I download firmware for navigation if I don't live in China

**Comment by HubrechtA** [comment]
I have Kona Petrol SX2 with ccNC interface and wonder if it's possible to enter the service menus and disable the multiple BEEP linked to the driver's assistance. Any other option is welcome.

**Comment by lollo109** [comment]
Hello,i have a Tucson 2018 restyling model and is there any way i can use google on it? I update it to the latest EU firmware gen 5 stamdard class. TLFLEUR.SOP.V140.241023.STD_M

**Comment by ayehia2005** [comment]
Hello ,, For new Hyundai cars like Palaside 2024 it won`t work ... there is a new file structure for the updates and it is compressed in .tar format.

is there anyway to crack it ? 
<img width="363" alt="2025-07-04_3-23-06" src="https://github.com/user-attachments/assets/0d4337ee-5293-4901-96c2-b704048659e1" />

**Comment by statist1c** [comment]
@rgerganov is it possible to change from KOR to EUR firmware? If yes please let me know how?

**Comment by 7ikw** [comment]
Can any one help me please with this se version NQ5.kor.s5w_M.v035.001.250507

**Comment by syboxez** [comment]
A couple of questions. I'm having the Manage Applications issue and I can't access ADB through a USB port using a type A to type A cable with the 5v line cut (same one that I use on my Android TV). My car also doesn't have WiFi. I have an ADPE.

I tried installing a terminal emulator ($hell) in the system apps, but /system/bin/amossu doesn't do anything and trying to use `pm` to install packages gives me a segfault. I also have a file manager (an old CyanogenMod one) installed, but I can't install APKs due to there not being an Install from Unknown Sources option.

My main question is how do I use ADB so I can fix all of these issues? I'm mainly trying to install RetroArch on my car so I can run DOOM on it. It would also be nice if I can properly root it instead of having to rely on amossu.

**Comment by thnayan755** [comment]
I imported a 2025 genesis g80 from Korea and the navigation system does not work in our Middle East region and I want to add the Middle East navigation system knowing the screen update version number rg3.kor.ccic2.001.006.250612 Please help me, thank you very much
![ba8e25de-ba47-4050-a8a1-8848e551692a](https://github.com/user-attachments/assets/94052fc3-931e-491f-b6ed-c1fe456904a9)

**Comment by maciejkoska2-png** [comment]
Hi, 

I have Hyundai Palisade 2023 LX2_PE.USA.S5W_M.V012 with the newer firmware format (no update_package.zip, instead new_system_package.tar with enc_system.ext4 files prefixed with enc_ instead of encrypt_).

The AES key from your blog post doesn't work for this format. Do you have any idea how to decrypt enc_system.ext4 for this newer firmware version? I've confirmed the file is 2GB and properly extracted from the tar archive.
Thanks!

**Comment by beren12** [comment]
Do you know of any archives of old firmware versions? I'm trying to find something in it for my 21 kona ev. so a 2020-2022 firmware may have it

**Comment by inquam** [comment]
Seems it is running android. apk installation from usb, Android Auto like experience, Youtube, Comma.ai (OpenPilot, SunnyPilot apps directly on the dash etc) would be cool. But perhaps the SoC is way to weak?
